platform/gtk4/cog-platform-gtk4.c

```diff
@@ -340,6 +340,18 @@ on_scroll(GtkEventControllerScroll* controller, double dx,
     return TRUE;
 }
 
+static gboolean
+is_pressed_key(guint keyval)
+{
+    static bool initialized = false;
+
+    if (!initialized) {
+        GUniqueOutPtr<GdkKeymapKey> keys;
```

Comment:
This won't work on a plain C source. It looks like you were picking from C++ code from elsewhere, so this will need adapting to compile as C99.